emulator.c

```diff
@@ -81,8 +92,18 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
         break;
       }
     case 0x19: // NOLINT
-      printf("DAD D");
-      break;
+      {        // DAD D
+        uint16_t sum
+            = (cpu->h << 8) // NOLINT
+              | (cpu->l);   // convert reg pair h,l to 16 bit int // NOLINT
+        sum = sum + cpu->d;
+
+        cpu->h = sum >> 8;                   // NOLINT
+        cpu->l = sum & 255;                  // NOLINT
```

Comment:
I think this is similar to what we talked about last night about one of your other instructions. I'm not sure the & is correctly preserving the LSB of sum here...

Reply:
Maybe cpu -> l = (uint8_t) sum & 0x00FF would be clearer?